src/Workspaces/CoreTest/CodeStyle/NamingStylePreferencesUpgradeTests.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests.CodeStyle
+{
+    public class NamingStylePreferencesUpgradeTests
```

Comment:
❗️ Add a test verifying that an attempt to load preferences with a version higher than the maximum known version simply causes the preferences to be ignored.

Reply:
Currently those preferences get reset (which is probably not a very good experience, which is why I was asking about this before). Could you please clarify what you mean by "ignored"? There has to be some set of preferences that actually gets loaded. What would that be if not the default preferences?

Reply:
> Currently those preferences get reset 

Yes, I expect a reset to default for this case.

Reply:
added in 014dadde7e4d7264b8cd6f5506718044ea0283db